File: Navtrack.Core/ClientApp/src/apis/types/asset/AssetModel.ts
import { LocationModel, DefaultLocationModel } from "../location/LocationModel";
import { DeviceModel, DefaultDeviceModel } from "../device/DeviceModel";

export type AssetModel = {
  id: number;
  name: string;
  activeDevice: DeviceModel;
  devices: DeviceModel[];
  location: LocationModel;
};

export const DefaultAssetModel: AssetModel = {
  id: 0,
  name: "",
  activeDevice: DefaultDeviceModel,
  location: DefaultLocationModel,
  devices: []
};

File: backend/Navtrack.Api.Model/Assets/AssetModel.cs
using System.ComponentModel.DataAnnotations;
using Navtrack.Api.Model.Messages;
using DeviceModel = Navtrack.Api.Model.Devices.DeviceModel;

namespace Navtrack.Api.Model.Assets;

public class AssetModel
{
    [Required]
    public string Id { get; set; }

    [Required]
    public string Name { get; set; }
    
    [Required]
    public string OrganizationId { get; set; }

    [Required]
    public bool Online { get; set; }

    [Required]
    public int MaxSpeed => 400; // TODO update this property

    public DeviceMessageModel? LastMessage { get; set; }
    public DeviceMessageModel? LastPositionMessage { get; set; }

    public DeviceModel? Device { get; set; }
}